src/org/rascalmpl/interpreter/env/GlobalEnvironment.java

```diff
@@ -95,6 +95,7 @@ public void registerSourceResolver(String scheme, ICallableValue function) {
 	 */
 	public ModuleEnvironment addModule(ModuleEnvironment mod) {
 		assert mod != null;
+		clearLookupChaches();
```

Comment:
This will clear caches if someone imports a new module in the REPL, but that can not have an effect on the cache consistency IMHO. Also during module loading and global variable initialization we will loose the cache all the time while new modules are being loaded. 

Perhaps this one is not necessary?

Reply:
I wanted to be extra sure. Especially due to cycles and everything. Just clear the caches a bit more aggressively, at the cost of some perfhit after an import.